lib/db/mem.js

```diff
@@ -1333,23 +1340,25 @@ module.exports = function (log, error) {
           throw error.notFound()
         }
 
-        let foundCode, foundIndex
-        for (let i = 0; i < codes.length; i++) {
-          const code = codes[i]
-          if (codeHash === code.codeHash.toString('hex')) {
-            foundCode = code
-            foundIndex = i
-            break
-          }
-        }
+        const hashes = codes.map((item, index) => {
+          return dbUtil.compareHashScrypt(code, item.codeHash, item.salt)
+            .then((equals) => {
+              return {item, index, equals}
+            })
+        })
 
-        if (! foundCode) {
-          throw error.notFound()
-        }
+        // Filter to only matching hashes
+        return P.filter(hashes, item => item.equals)
+          .map((item) => item.index)
```

Comment:
Could we eliminate this extra `.map` loop if the code below referred to `result[0].index` instead?

Reply:
(sorry, realise I'm bikeshedding the mem db here, which is kind of pointless of me)